tokio/src/runtime/handle.rs

```diff
@@ -202,7 +202,105 @@ impl Handle {
         handle
     }
 
-    /// TODO: write docs if this is a good direction
+    /// Run a future to completion on this `Handle`'s associated `Runtime`.
+    ///
+    /// This runs the given future on the runtime, blocking until it is
+    /// complete, and yielding its resolved result. Any tasks or timers which
+    /// the future spawns internally will be executed on the runtime.
+    ///
+    /// The behavior for multi threaded vs current thread schedulers is the same
+    /// as [`Runtime::block_on`]. See the docs of [`Runtime::block_on`] for more
+    /// details.
```

Comment:
Is this true? Will it drive the IO driver when its the only call to block_on on a current_thread runtime?

Reply:
I remember checking that some time ago. But now I'm not sure. I'll have a look and perhaps add a test for it.

Reply:
Ah wait sorry. I commented on the wrong thing.

> Is this true? Will it drive the IO driver when its the only call to block_on on a current_thread runtime?

I'm not sure about this. Don't know enough about the internals to say 🤷 Do you know?

Reply:
The `Handle::block_on` in Tokio 0.2 didn't do this, but I don't know if that has changed since then. To be clear, it is plausible that it could have changed as part of `Runtime::block_on` being changed from `&mut self` to `&self`. But I don't know.

@carllerche do you know?

Reply:
`Handle::block_on` and `Runtime::block_on` are slightly different. `Handle::block_on` **will not** drive the runtime where as `Runtime::block_on` **may** drive the runtime. Currently, this only happens with the `current_thread` scheduler.

Reply:
Alright. Is this something you think is worth mentioning or should I just cut out this bit?

Reply:
Yes, we should mention this.

Reply:
I've been thinking about this but sure how to word it accurately.

Reply:
How about this?

> When this is used on a current_thread runtime, only the `Runtime::block_on` method can drive the IO and timer drivers, but the `Handle::block_on` method cannot drive them. This means that, when using this method on a current_thread runtime, anything that relies on IO or timers will not work unless there is another thread currently calling `Runtime::block_on` on the same runtime.

Reply:
I am applying ^^ to the PR so we can merge :)

Reply:
Actually, it seems like I can't push to your PR :'(